app/docs/md/patterns/rendering-markdown.md

```diff
@@ -0,0 +1,197 @@
+
+In our `app/api/markdown/$$.mjs` file, we'll need to import a few packages and instantiate Arcdown.
+
+```javascript
+import { readFileSync } from 'fs'
+import { URL } from 'url'
+import { Arcdown } from 'arcdown'
+const arcdown = new Arcdown()
+```
+
+Then in our `get` function we'll need to figure out which file the user has requested. To do that we'll determine the document path from the incoming request.
+
+```javascript
+ const { path: activePath } = req
+ let docPath = activePath.replace(/^\/?docs\//, '') || 'index'
+ if (docPath.endsWith('/')) {
+   docPath += 'index' // trailing slash == index.md file
+ }
+```
+
+Next we'll use the `docPath` to read our markdown file from the `app/markdown` folder.
+
+```javascript
+ const docURL = new URL(`../../${docPath}.md`, import.meta.url)
+ const docMarkdown = readFileSync(docURL.pathname, 'utf-8')
+```
+
+Once we have the markdown string, we can transform it into HTML using Arcdown and add it to the `store`.
+
+```javascript
+ const doc = await arcdown.render(docMarkdown)
+ return {
+   json: { doc }
+ }
+```
+
+The entire function looks like this:
+
+```javascript
+// app/api/markdown/$$.mjs
+import { readFileSync } from 'fs'
+import { URL } from 'url'
+import { Arcdown } from 'arcdown'
+const arcdown = new Arcdown()
+
+export async function get (req) {
+ // Get requested path
+ const { path: activePath } = req
+ let docPath = activePath.replace(/^\/?docs\//, '') || 'index'
+ if (docPath.endsWith('/')) {
+   docPath += 'index' // trailing slash == index.md file
+ }
+
+ // Read markdown file
+ const docURL = new URL(`../../${docPath}.md`, import.meta.url)
+ const docMarkdown = readFileSync(docURL.pathname, 'utf-8')
+
+ // Convert to HTML and add to store
+ const doc = await arcdown.render(docMarkdown)
+ return {
+   json: { doc }
+ }
+}
+```
+
+> **Note:** I'm omitting a lot of error checking for the sake of brevity but check out this [hardened example](https://github.com/enhance-dev/enhance.dev/blob/main/app/api/docs/%24%24.mjs) for more details on how to handle 404's.
+
+## Displaying markdown
+
+Now that we have successfully transformed our markdown into HTML let's go about displaying it in a page. First, we'll create a new catch-all page under `app/pages/markdown/$$.html` and a new web component to display the markdown at `app/elements/doc-content.mjs` which you can do manually or by running the Begin CLI commands:
+
+```bash
+enhance gen page --path 'markdown/$$'
+enhance gen element --name doc-content
+```
+
+Our project structure now looks like this:
+
+```
+app
+├── api
+|   └── markdown
+|       └── $$.mjs
+├── elements
+|   └── doc-content.mjs
+├── markdown
+|   └── example.md
+└── pages
+    ├── markdown
+    |   └── $$.mjs
+    └── index.html
+```
+
+Our `app/pages/markdown/$$.html` file will look really simple as we will just hand everything over to our web component to handle. Feel free to make this page more advanced by adding headers, footers, etc. as required by your design.
```

Comment:
Can probably trim a fair bit here this is evidently just an example; I reckon folks will know they can add more if they want to ;)

```suggestion
For the purposes of this example, we'll keep our `app/pages/markdown/$$.html` simple by including only the web component we just created:
```